tuner/tuner/candidate_gen.py

```diff
@@ -40,36 +40,46 @@
 
 
 def apply_configuration(
-    template: list[str], configuration: Configuration, tile_sizes: list[int]
+    template: list[str],
+    configuration: Configuration,
+    workgroup_sizes: list[int],
+    reduction_sizes: list[int],
```

Comment:
Could we extract workgroup and tile sizes from the `configuration` directly and remove these two function arguments?

Reply:
yeah, good catch!

Reply:
Oh, actually no. The tile sizes from the configuration are not used directly https://github.com/nod-ai/shark-ai/blob/main/tuner/tuner/dispatch_parser.py#L41, so these two function arguments cannot be removed.

Reply:
This definitely should be removed eventually, but we don't have to put everything in the same PR. Could you add a TODO so that we don't forget about this?

Reply:
Sure, will do.